src/spurt/io/_three_d.py

```diff
@@ -0,0 +1,263 @@
+
+    @property
+    def dtype(self) -> DTypeLike:
+        return self._data.dtype
+
+    @property
+    def ndim(self) -> int:
+        return 2
+
+    @property
+    def time_dim(self) -> int:
+        return int(self._time_dim > 0)
+
+    @property
+    def space_dim(self) -> int:
+        return int(self._time_dim == 0)
+
+    @property
+    def _space_shape(self) -> tuple[int, ...]:
+        return tuple(
+            ss for ii, ss in enumerate(self._data.shape) if ii != self._time_dim
+        )
+
+    @property
+    def _space_size(self) -> int:
+        return int(np.prod(self._space_shape))
+
+    @property
+    def _time_size(self) -> int:
+        return self._data.shape[self._time_dim]
+
+    @property
+    def shape(self) -> tuple[int, ...]:
+        if self.time_dim:
+            return (self._space_size, self._time_size)
+
+        return (self._time_size, self._space_size)
+
+    @property
+    def xcoords(self) -> ArrayLike:
+        """Return the x-coordinate of the points as an array.
+
+        This is to mimic an irregular grid.
+        """
+        return np.arange(np.prod(self._data.shape), dtype=int) % self._space_shape[1]
+
+    @property
+    def ycoords(self) -> ArrayLike:
+        """Return the y-coordinate of the points as an array.
+
+        This is to mimic an irregular grid.
+        """
+        return np.arange(np.prod(self._data.shape), dtype=int) // self._space_shape[1]
+
+    def get_time_slice(self, key: int) -> ArrayLike:
+        """Read a block of data in time.
+
+        Only makes a copy if absolutely needed.
+        """
+        ind = tuple(
+            slice(None) if ii != self._time_dim else int(key)
+            for ii in range(self._data.ndim)
+        )
+        return self._data[ind].ravel()  # type: ignore[arg-type, index]
+
+    def get_spatial_slice(self, key: int) -> ArrayLike:
+        """Read a block of data in space.
+
+        Only makes a copy if absolutely needed.
+        """
+        ind = list(np.unravel_index(key, self._space_shape))
+        ind.insert(self._time_dim, slice(None))
+
+        return self._data[tuple(ind)].ravel()  # type: ignore[arg-type, index]
+
+
+class Irreg3DInput(InputStackInterface):
+    """A single numpy 2D array as input.
+
+    One axis represents time and the other some flattened representation
+    in space. `time_dim` is an optional input to indicate time axis.
+    """
+
+    def __init__(
+        self,
+        data: InputInterface,
+        xy: InputInterface,
+        time_dim: int = 0,
+    ):
+        """Initialize a stack interface for a 2D numpy array.
+
+        One axis represents time axis and the other spatial axis.
+
+        Parameters
+        ----------
+        data: np.ndarray
+            Numpy 2D array as input
+
+        xy: np.ndarray
+            Numpy npts x 2 array as input
+
+        time_dim: int, optional
+            Axis of the numpy array that represents the time dimension
+        """
```

Comment:
Let's remove references to "numpy" now that this interface is generic.

```suggestion
        """Initialize a stack interface for a 2D array.

        One axis represents time axis and the other spatial axis.

        Parameters
        ----------
        data: InputInterface
            2D array as input

        xy: InputInterface
            npts x 2 array as input

        time_dim: int, optional
            Axis of the array that represents the time dimension
        """
```